### DISCUSSION TITLE
Could not load namespace file

### DISCUSSION THREAD

**Discussion by richterich** [ivanhofer/typesafe-i18n#491] [Q&A] [answered]
Hello! I bump into a problem with the generator. I used sveltekit with vite and I added the typesafe-i18n library to my project. It created a .typesafe-i18n.json config file with the content:
```json
{
   "adapter": "svelte",
   "$schema": "https://unpkg.com/typesafe-i18n@5.15.0/schema/typesafe-i18n.json",
   "outputFormat": "JavaScript"
}
```
My problem is that the generator gave the error message when I created a namespace file `src/i18n/en/terms/index.js`
**Terminal Output**
```terminal
file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20499
var parseRawText = (rawText, optimize = true, firstKey = "", lastKey = "") => rawText.split(REGEX_BRACKETS_SPLIT).map((part) => {
                   ^

RangeError: Maximum call stack size exceeded
    at parseRawText (file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20499:20)
    at parseTranslationEntry (file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20552:23)
    at file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20538:12
    at Array.map (<anonymous>)
    at parseDictionary (file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20536:149)
    at file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20540:19
    at Array.map (<anonymous>)
    at parseDictionary (file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20536:149)
    at file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20540:19
    at Array.map (<anonymous>)
```
When I changed the file extension to `ts` the error was gone and the generator generate the namespace folder structure and file `index.js` (the extension of the generated file was `js`) in another locale folder (`ua`). It also generated new `NamespaceTermsTranslation` in the `i18n-types.d.ts` and other auto-generated code, but it gave a message that `could not load namespace file en/terms`

As a result, that message does not crash the generator, but it misses type generation for the namespace locale file. Could you please help me solve this problem?

**Note:** the same for the file with the extension `mjs`.

**Comment by ivanhofer** [comment, marked as answer]
Hi @richterich, can you please create a demo repository where I can reprodue the issue?

**Reply by richterich** [reply]
Hi, @ivanhofer! It here https://github.com/richterich/typesafe-i18n-load-namespaces

**Reply by ivanhofer** [reply]
Thanks for the gif in the repo!

From what I can see there (without actually checking your repository) is:
 - I think the generator runs into a weird loop if the default export is missing. Please try to create the `en/terms/index.js` file with a default export and run the generator again. Let me know if it works :)
 - `.ts` files are not parsed when , so the second part of the gif looks like "works as intended"

**Reply by richterich** [reply]
I created the `en/terms/index.js` file with a default export but it got crashed.
```terminal
file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20499
var parseRawText = (rawText, optimize = true, firstKey = "", lastKey = "") => rawText.split(REGEX_BRACKETS_SPLIT).map((part) => {
                   ^

RangeError: Maximum call stack size exceeded
    at parseRawText (file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20499:20)
    at parseTranslationEntry (file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20552:23)
    at file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20538:12
    at Array.map (<anonymous>)
    at parseDictionary (file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20536:149)
    at file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20540:19
    at Array.map (<anonymous>)
    at parseDictionary (file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20536:149)
    at file:///home/alex/Sources/repos/my-website/node_modules/typesafe-i18n/cli/typesafe-i18n.mjs:20540:19
    at Array.map (<anonymous>)

```

**Reply by ivanhofer** [reply]
alright, I can take a look at this in the coming days

**Reply by ivanhofer** [reply]
I have found and fixed the issue. Please try it again with version `5.16.1`

**Reply by richterich** [reply]
Yes! It works nicely in version `5.16.1`
Thank You!
